### PR TITLE
Update QuickChick CI

### PR DESCRIPTION
**Pull request by Lysxia**

### CODE REVIEW

**Review by liyishuai — APPROVED** [review]

File: dev/ci/ci-quickchick.sh [hunk 1]
```diff
@@ -10,6 +10,6 @@ git_download quickchick
 if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
 
 ( cd "${CI_BUILD_DIR}/quickchick"
-  make
-  make install
+  dune build -p coq-quickchick @install
+  dune install -p coq-quickchick --prefix=$CI_INSTALL_DIR
```
Comment (SkySkimmer) [inline dev/ci/ci-quickchick.sh:14]:
```suggestion
  dune build --root . -p coq-quickchick @install
  dune install coq-quickchick --prefix=$CI_INSTALL_DIR
```
IIRC this is how using dune in ci works

**Comment by SkySkimmer** [comment]
@coqbot run full ci

File: dev/ci/ci-quickchick.sh [hunk 2]
```diff
@@ -10,6 +10,6 @@ git_download quickchick
 if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
 
 ( cd "${CI_BUILD_DIR}/quickchick"
-  make
-  make install
+  dune build --root . -p coq-quickchick @install
```
Comment (SkySkimmer) [inline dev/ci/ci-quickchick.sh:13]: actually should be 
```suggestion
  dune build -p coq-quickchick @install
```
(-p includes -root .)
sorry for previous incorrect suggestion

**Comment by SkySkimmer** [comment]
quickchick_test failed
Also it seems to recompile quickchick, possibly because the main job is in release mode (from -p)?

**Comment by SkySkimmer** [comment]
Seems to work
@coqbot merge now

**Pull request merged**